shogun/classifier/svm/WDSVMOcas.cpp

```diff
@@ -143,16 +143,16 @@ SGVector<float64_t> CWDSVMOcas::apply_get_outputs(CFeatures* data)
 	return outputs;
 }
 
-int32_t CWDSVMOcas::set_wd_weights()
+index_t CWDSVMOcas::set_wd_weights()
 {
 	ASSERT(degree>0 && degree<=8)
 	SG_FREE(wd_weights);
 	wd_weights=SG_MALLOC(float32_t, degree);
 	SG_FREE(w_offsets);
-	w_offsets=SG_MALLOC(int32_t, degree);
-	int32_t w_dim_single_c=0;
+	w_offsets=SG_MALLOC(index_t, degree);
+	index_t w_dim_single_c=0;
 
-	for (int32_t i=0; i<degree; i++)
+	for (index_t i=0; i<degree; i++)
```

Comment:
is the type of degree `index_t`? as it'd be better to use `auto i` but for that degree needs to be index_t